lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1642,6 +1642,14 @@ protected function _validateAndCompleteAssociationMapping(array $mapping)
 
         if ($this->isTypedProperty($mapping['fieldName'])) {
             $mapping = $this->validateAndCompleteTypedAssociationMapping($mapping);
+        } elseif (isset($mapping['joinColumns'])) {
+            foreach ($mapping['joinColumns'] as &$joinColumn) {
```

Comment:
I believe this will not work here, because `_validateAndCompleteAssociationMapping` is called before the code for setting default joinColumns is called in `_validateAndCompleteOneToOneMapping`.

I believe, you need to put this code into `_validateAndCompleteOneToOneMapping`, it is also used for ManyToOne, and those two assocations are the only two that we are interested in completing anyways.

Otherwise it will not work for the case where you only specify `@ManyToOne` but not a corresponding `@JoinColumn`.

Reply:
You are right and adding tests for `@ManyToOne` without `joinColumn` uncovered it. I have fixed it exactly as suggested.